### ISSUE TITLE
Unable to log in with basic-auth due to GDPR mode

### ISSUE THREAD

**Issue by EpicWink** [pycontribs/jira#935]
I'm unable to log in with basic-auth due to GDPR strict mode which is now enforced.

### Description
When trying to authenticate with JIRA during first request, I get an error saying I can't use username.

### Reproduction
```python
jac = jira.JIRA(
    server="https://<myorg>.atlassian.net/",
    basic_auth=(account_id, api_token),
    options={"rest_api_version": 3},
)
users = jac.search_users("example-user")
```

I've tried passing in username, display-name and email instead of `account_id`, and password instead of `api_token`, and still I get an error

### Expected behaviour
Successful authentication.

### Stack Trace
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "~/.local/lib/python3.7/site-packages/jira/client.py", line 2840, in search_users
    return self._fetch_pages(User, None, "user/search", startAt, maxResults, params)
  File "~/.local/lib/python3.7/site-packages/jira/client.py", line 640, in _fetch_pages
    resource = self._get_json(request_path, params=page_params, base=base)
  File "~/.local/lib/python3.7/site-packages/jira/client.py", line 3129, in _get_json
    r = self._session.get(url, params=params)
  File "~/.local/lib/python3.7/site-packages/jira/resilientsession.py", line 173, in get
    return self.__verb("GET", url, **kwargs)
  File "~/.local/lib/python3.7/site-packages/jira/resilientsession.py", line 169, in __verb
    raise_on_error(response, verb=verb, **kwargs)
  File "~/.local/lib/python3.7/site-packages/jira/resilientsession.py", line 55, in raise_on_error
    r.status_code, error, r.url, request=request, response=r, **kwargs
jira.exceptions.JIRAError: JiraError HTTP 400 url: https://<myorg>.atlassian.net/rest/api/3/user/search?username=<api_token>&includeActive=True&includeInactive=False&maxResults=50
    text: The query parameter 'username' is not supported in GDPR strict mode.
    
    response headers = {'Server': 'AtlassianProxy/1.15.8.1', 'Cache-Control': 'no-cache, no-store, no-transform', 'Content-Type': 'application/json;charset=UTF-8', 'Strict-Transport-Security': 'max-age=315360000; includeSubDomains; preload', 'Date': 'Wed, 06 May 2020 03:49:32 GMT', 'ATL-TraceId': '82464007f0031d21', 'X-AREQUESTID': 'e4be7405-6a4e-4046-bce4-276a1ec43888', 'X-XSS-Protection': '1; mode=block', 'Transfer-Encoding': 'chunked', 'Timing-Allow-Origin': '*', 'X-Content-Type-Options': 'nosniff', 'Micros-Issuer': 'micros/edge-authenticator', 'Connection': 'close', 'Expect-CT': 'report-uri="https://web-security-reports.services.atlassian.com/expect-ct-report/global-proxy", enforce, max-age=86400'}
    response text = {"errorMessages":["The query parameter 'username' is not supported in GDPR strict mode."],"errors":{}}
```

### Environment
Python Interpreter: Python 3.7.5
jira-python: 2.0.0 and 3.0a2
OS: Ubuntu 19.10
requests: 2.23.0

**Comment by eplodn** [comment]
It looks like you are able to authenticate but unable to search for users?

Addressed by https://github.com/pycontribs/jira/pull/941

**Comment by EpicWink** [comment]
@eplodn you're right, if I replace
```python
users = jac.search_users("example-user")
```

with
```python
jac._fetch_pages(jira.User, None, "user/search", 0, 50, {"query": "example-user"})
```

Then everything works fine

**Comment by adehad** [comment]
The bug in searching users has been addressed in: #927

**Comment by cloutsocks** [comment]
this still seems broken for me, even when on master
